packages/uui-popover-container/lib/uui-popover-container.element.ts

```diff
@@ -0,0 +1,285 @@
+import { defineElement } from '@umbraco-ui/uui-base/lib/registration';
+import { css, html, LitElement } from 'lit';
+import { property, state } from 'lit/decorators.js';
+
+export type PopoverContainerPlacement =
+  | 'auto'
+  | 'top'
+  | 'top-start'
+  | 'top-end'
+  | 'bottom'
+  | 'bottom-start'
+  | 'bottom-end'
+  | 'right'
+  | 'right-start'
+  | 'right-end'
+  | 'left'
+  | 'left-start'
+  | 'left-end';
+
+/**
+ * @element uui-popover-container
+ */
+@defineElement('uui-popover-container')
+export class UUIPopoverContainerElement extends LitElement {
+  /**
+   * Set the distance between popover-modal and trigger.
+   * @type {number}
+   * @attr disabled
+   * @default false
+   */
+  @property({ type: Number })
+  margin = 0;
+
+  /**
+   * Define the placement of the popover-modal.
+   * @attr placement
+   * @default 'bottom-start'
+   */
+  @property({ type: String, reflect: true })
+  get placement(): PopoverContainerPlacement {
+    return this._placement;
+  }
+  set placement(newValue: PopoverContainerPlacement) {
+    this._placement = newValue;
+    this.#updatePadding();
+    this.#updatePosition();
+  }
+
+  @state()
+  _placement: PopoverContainerPlacement = 'bottom-start';
+
+  #target: HTMLElement | null = null;
+
+  connectedCallback(): void {
+    super.connectedCallback();
+
+    // CHECK BROWSER SUPPORT
+    if (!HTMLElement.prototype.hasOwnProperty('popover')) {
+      alert(
+        'Browser does not support popovers. Check the docs for info on how to enable: https://developer.mozilla.org/en-US/docs/Web/API/Popover_API'
+      );
+      return;
+    }
+
+    this.addEventListener('beforetoggle', this.#beforeToggle);
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+
+    this.removeEventListener('beforetoggle', this.#beforeToggle);
+  }
+
+  #updatePadding = () => {
+    const oppositeSides: Record<string, string> = {
+      top: 'bottom',
+      bottom: 'top',
+      left: 'right',
+      right: 'left',
+    };
+
+    // find the side excluding start/end
+    let side = this._placement.split('-')[0];
+    // find the opposite side
+    side = oppositeSides[side] || side;
+    // capitalize the side
+    side = side.charAt(0).toUpperCase() + side.slice(1);
+
+    const paddingSide = `padding${side}`;
+    this.style.padding = '0';
+    (this.style as any)[paddingSide] = `10px`;
+  };
+
+  #beforeToggle = async (event: any) => {
+    this.#target = this.#findAncestorWithAttribute(
+      this,
+      'popovertarget',
+      this.id
+    );
+
+    // Dispatch a custom event that can be listened to by the popover target.
+    // Mostly used for UUIButton.
```

Comment:
With this I'm starting to think if we need to make this "popover-button-feature" as a Mixin, so we also can make this work for uui-tab?